### ISSUE TITLE
colour scale graph in R-Instat?

### ISSUE THREAD

**Issue by rdstern** [africanmathsinitiative/R-Instat#8623]
I have now managed to get the following graph:
<img width="516" alt="image" src="https://github.com/africanmathsinitiative/R-Instat/assets/11226469/c36bd60a-bcb5-4b85-9e3c-b10001d8d782">

I used to general graphics dialog, plus the existing colours from the sub-dialog.

That gave me the following code:
```
# Dialog: Graphics

us_contagious_diseases <- data_book$get_data_frame(data_name="us_contagious_diseases")
last_graph <- ggplot2::ggplot(data=us_contagious_diseases, mapping=ggplot2::aes(x=year, y=state, fill=rate)) + ggplot2::geom_tile() + ggplot2::geom_vline(xintercept=1963.00) + viridis::scale_fill_viridis(option="inferno") + theme_grey()
data_book$add_object(data_name="us_contagious_diseases", object_name="last_graph", object_type_label="graph", object_format="image", object=check_graph(graph_object=last_graph))
data_book$get_object_data(data_name="us_contagious_diseases", object_name="last_graph", as_file=TRUE)
rm(list=c("last_graph", "us_contagious_diseases"))
```

Then I replaced the existing viridis function with the one from the Data Science book!  This gives:

```
# Dialog: Graphics

library(RColorBrewer)

us_contagious_diseases <- data_book$get_data_frame(data_name="us_contagious_diseases")
last_graph <- ggplot2::ggplot(data=us_contagious_diseases, mapping=ggplot2::aes(x=year, y=state, fill=rate)) + ggplot2::geom_tile() + ggplot2::geom_vline(xintercept=1963.00) + scale_fill_gradientn(colors = brewer.pal(9, "Reds"), trans = "sqrt")+ theme_grey()
data_book$add_object(data_name="us_contagious_diseases", object_name="last_graph", object_type_label="graph", object_format="image", object=check_graph(graph_object=last_graph))
data_book$get_object_data(data_name="us_contagious_diseases", object_name="last_graph", as_file=TRUE)
rm(list=c("last_graph", "us_contagious_diseases"))
```

@MeSophie and @lilyclements will this, or something similar, be possible with the new colour option, or is something more needed?

**Comment by MeSophie** [comment]
![image](https://github.com/africanmathsinitiative/R-Instat/assets/106779233/e13e1c83-169c-49a4-aebc-b5e7ff299819)

@rdstern  means that I will add geom-vline on the list of geoms in General Graphics dialog?
Also @rdstern , @lilyclements  how can I add the  `scale_fill_gradientn ` on the actual colour tab design? I was thinking to add one more button labelled `RColorbrewer` on the list of the actual buttons.

**Comment by lilyclements** [comment]
> Also @rdstern , @lilyclements how can I add the scale_fill_gradientn  on the actual colour tab design? I was thinking to add one more button labelled RColorbrewer on the list of the actual buttons.

@MeSophie it could be added there, or in the meantime, we can get the scale in a different way. I think something like this should work -

```
scale_colour_gradient(low = "white", high = "red", trans = "sqrt")
```

**Comment by MeSophie** [comment]
> > Also @rdstern , @lilyclements how can I add the scale_fill_gradientn  on the actual colour tab design? I was thinking to add one more button labelled RColorbrewer on the list of the actual buttons.
> 
> @MeSophie it could be added there, or in the meantime, we can get the scale in a different way. I think something like this should work -
> 
> ```
> scale_colour_gradient(low = "white", high = "red", trans = "sqrt")
> ```

@lilyclements Thank you for this information. But I still need you point of view about the design.  `scale_colour_gradient/scale_colour_gradientn` are comming from ggplot2 package where am I going to include the function on the design? Can I add a new button after Ggthermes ?

**Comment by lilyclements** [comment]
@MeSophie thank you for clarifying your question. I think these could be added as a new button called "Gradient" or something? @MeSophie @rdstern what do you think?

**Comment by MeSophie** [comment]
> @MeSophie thank you for clarifying your question. I think these could be added as a new button called "Gradient" or something? @MeSophie @rdstern what do you think?

@lilyclements I agree. So Now @rdstern  can I add it now on PR #8613 or I need  to wait it to be merge first?
